src/core/server/http/integration_tests/core_services.test.ts

```diff
@@ -410,5 +414,31 @@ describe('http service', () => {
       const [, , clientHeaders] = client;
       expect(clientHeaders).toEqual({ authorization: authorizationHeader });
     });
+
+    it('forwards 401 errors returned from elasticsearch', async () => {
+      const { http } = await root.setup();
+      const { createRouter } = http;
+
+      const authenticationError = LegacyElasticsearchErrorHelpers.decorateNotAuthorizedError(
+        new (esErrors.AuthenticationException as any)('Authentication Exception', {
+          body: { error: { header: { 'WWW-Authenticate': 'authenticate header' } } },
+          statusCode: 401,
+        })
+      );
```

Comment:
🙈 but as the `ClusterClient` is already mocked in the test suite, I couldn't 'demock' it to throw the error directly from the `Client` itself. Also the behavior from the client is tested in `cluster_client.test.ts` so I guess this is fine.

Reply:
you could throw this exception from a route handler

Reply:
Yea, I could throw the same error directly from the handler instead of mocking the return of `callAsCurrentUser`, but forging the error would still be as ugly.